client/me/purchases/controller.jsx

```diff
@@ -156,7 +156,7 @@ export default {
 		);
 	},
 
-	list() {
+	list( context ) {
```

Comment:
I try to avoid using global notices when possible, because the `notices` module can't really be used in a component's render(), and that is where I would expect to find a notice's message and markup. The notice message would live in `PurchasesController` in your example, yeah?

In this case, I also think the fact that the notice does not persist when the user navigates to a new page is preferable, but I might be in the minority with that. There is also the trade-off you pointed out: the notice exists in the browser history this way, which isn't ideal.

If a global notice would provide a better UX than the current implementation (the notice is associated with `/purchases/cancel-success`, so the notice disappears when the user navigates to a new page, but would return if the user presses `back`) I'm happy to make the change. @gziolo @mikeshelton1503 What do you think?